### PR TITLE
Android performance/state management refactor

### PR DESCRIPTION
**Pull request by JackVCurtis**
### Summary

_Ticket:_ [Address jankiness / lag in UI](https://app.asana.com/0/1205732265579288/1208725410378688/f)

What is this PR for?

This PR addresses consistent "jank" (lag, stuttering) in the app due to an overloaded UI thread.

- [x] If you added any user facing strings on iOS, are they included in Localizable.xcstrings?

### Testing

What testing have you done?

Manual testing for the expected improvements in performance, unit testing for the code changes related to state management.

### CODE REVIEW

File: androidApp/src/androidTest/java/com/mbta/tid/mbta_app/android/state/SubscribeToPredictionsTest.kt [hunk 1]
```diff
@@ -51,20 +49,16 @@ class SubscribeToPredictionsTest {
...
 
                 override var lastUpdated: Instant? = null
 
                 override fun shouldForgetPredictions(predictionCount: Int) = false
 
                 override fun disconnect() {
-                    check(isConnected) { "called disconnect when not connected" }
-                    isConnected = false
-                    launch { disconnectChannel.send(Unit) }
+                    /* null-op */
```
Comment (KaylaBrady) [inline androidApp/src/androidTest/java/com/mbta/tid/mbta_app/android/state/SubscribeToPredictionsTest.kt:61]: question: Is there still a meaningful alternative way to test that disconnect is called as expected? Could we invoke `onCleared` directly in a test?

**Review by KaylaBrady — APPROVED** [review]
Nice! Just a few small testing questions

File: androidApp/src/main/java/com/mbta/tid/mbta_app/android/state/subscribeToPredictions.kt [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            timerViewModel.timerFlow.collect {
+                synchronized(predictions) { predictions.notifyAll() }
+            }
+        }
+    }
+
+    override fun onCleared() {
+        super.onCleared()
+        predictionsRepository.disconnect()
+    }
```
[the remark this answers is not in the document]
Comment (JackVCurtis) [reply androidApp/src/main/java/com/mbta/tid/mbta_app/android/state/subscribeToPredictions.kt:50]: I'm not sure! I'll test it, my hunch is that the view model will reinstantiate on return from backgrounding but I don't really know.

File: androidApp/src/androidTest/java/com/mbta/tid/mbta_app/android/state/SubscribeToPredictionsTest.kt (hunk 1 above)
Comment (JackVCurtis) [reply androidApp/src/androidTest/java/com/mbta/tid/mbta_app/android/state/SubscribeToPredictionsTest.kt:61]: It's looking like predictions is behaving correctly on backgrounding! I'm seeing a `SocketError` on background (after changing all the `rememberSaveable`s) but it doesn't crash. Predictions in the log start and stop as expected.

File: androidApp/src/main/java/com/mbta/tid/mbta_app/android/state/subscribeToPredictions.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (JackVCurtis) [reply androidApp/src/main/java/com/mbta/tid/mbta_app/android/state/subscribeToPredictions.kt:50]: It's looking like predictions is behaving correctly on backgrounding! I'm seeing a SocketError on background (after changing all the rememberSaveables) but it doesn't crash. Predictions in the log start and stop as expected. 

(Sorry replied to the wrong comment)

File: androidApp/src/androidTest/java/com/mbta/tid/mbta_app/android/state/SubscribeToPredictionsTest.kt (hunk 1 above)
Comment (JackVCurtis) [reply androidApp/src/androidTest/java/com/mbta/tid/mbta_app/android/state/SubscribeToPredictionsTest.kt]: Ugh that was a struggle but `onClear` functionality is tested now

**Pull request merged**